### PR TITLE
add Busmapping coinbase code

### PR DESCRIPTION
**Pull request by DreamWuGit**
this pr base on https://github.com/appliedzkp/zkevm-circuits/pull/259
Busmapping related files: 
bus-mapping/src/evm/opcodes/coinbase.rs
bus-mapping/src/evm/opcodes.rs

### CODE REVIEW

**Comment by HAOYUatHZ** [comment]
changed the base branch from `main` to `block_coinbase` to better read the diff

**Review by HAOYUatHZ — APPROVED** [review]

File: bus-mapping/src/evm/opcodes/coinbase.rs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        eth_types::ToWord,
+        evm::StackAddress,
+        mock,
+    };
+    use pretty_assertions::assert_eq;
+
+    #[test]
+    fn coinbase_opcode_impl() -> Result<(), Error> {
+        let code = bytecode! {
+            // PUSH1(0x1)
```
Comment (icemelon) [inline bus-mapping/src/evm/opcodes/coinbase.rs:44]: remove comments?

**Comment by icemelon** [comment]
probably just merge this PR into the coinbase circuit one since it's quite simple, and you can then change the test cases in coinbase circuit using the busmapping

File: bus-mapping/src/evm/opcodes/coinbase.rs (hunk 1 above)
Comment (DreamWuGit) [reply bus-mapping/src/evm/opcodes/coinbase.rs:44]: yes, will update

**Comment by DreamWuGit** [comment]
> probably just merge this PR into the coinbase circuit one since it's quite simple, and you can then change the test cases in coinbase circuit using the busmapping

PR# 259 is in review status, maybe change afterwards, For now,I tend to keep them seperated before merged to reduce variable dependency

File: bus-mapping/src/evm/opcodes/coinbase.rs (hunk 1 above)
Comment (DreamWuGit) [reply bus-mapping/src/evm/opcodes/coinbase.rs]: fixed in bd5d427

**Pull request merged**